sdk/messaging/azservicebus/messageSettler.go

```diff
@@ -22,27 +24,32 @@ type messageSettler struct {
 	notifySettleOnManagement func(message *ReceivedMessage)
 }
 
-func newMessageSettler(links internal.AMQPLinks, retryOptions RetryOptions) *messageSettler {
+func newMessageSettler(tracer tracing.Tracer, links internal.AMQPLinks, retryOptions RetryOptions) *messageSettler {
 	return &messageSettler{
+		tracer:                   tracer,
 		links:                    links,
 		retryOptions:             retryOptions,
 		notifySettleOnLink:       func(message *ReceivedMessage) {},
 		notifySettleOnManagement: func(message *ReceivedMessage) {},
 	}
 }
 
-func (s *messageSettler) settleWithRetries(ctx context.Context, settleFn func(receiver amqpwrap.AMQPReceiver, rpcLink amqpwrap.RPCLink) error) error {
+func (s *messageSettler) settleWithRetries(ctx context.Context, settleFn func(receiver amqpwrap.AMQPReceiver, rpcLink amqpwrap.RPCLink) error, to *tracing.StartSpanOptions) error {
 	if s == nil {
 		return internal.NewErrNonRetriable("messages that are received in `ReceiveModeReceiveAndDelete` mode are not settleable")
 	}
 
+	if to != nil {
```

Comment:
We are setting the tracer here so we don't hit a nil-pointer dereference in cases where messageSettler is set to nil in a `receive-and-delete` sender

Reply:
There might be a bug, but I wouldn't expect to call into this code _at all_ with a ReceiveAndDelete sender. 

I should just fix that bug instead of you adding workaround code here.